setup.cfg

```
@@ -26,8 +26,8 @@ include_package_data = True
 zip_safe = False
 install_requires =
     protobuf >= 4.22
-    grpcio >= 1.37.0
-    grpcio-status>=1.37.0
+    grpcio >= 1.67.0
```

@berndverst any objections on bumping the grpc library version? 
This has been updated in the durable task fork and it's a dependency here.

---

Version 1.37.0 is very old, it's from April 2021. I think we're safe to update.

---

This does not mean it would use 1.37 but that 1.37 would be possible to be used. Pip by default will always install the latest version matching the constraint. But if another library a customer is using required `grpcio==1.47.0` for example it would be possible for that specific version to be used.

In Python two versions cannot coexist unlike in other languages. So it's really really important not to force the minimum version to something recent unless this is absolutely necessary.

---

```suggestion
    grpcio >= 1.42.0
```

Short of auditing the Dapr community to figure out what libraries everyone uses and what the dependencies are pinning... here is a quick thing I identified. This is evidence of at least 1.42.0 being used for something.

https://github.com/tensorflow/tensorflow/blob/8fd1cd0826ef2b12c00540f3acc5a6e3c249f5b2/ci/official/containers/linux_arm64/devel.usertools/test.requirements.txt#L2


If as a project we want to say that we do not care about ML users - or that ML users must be on the latest official `grpcio` packages (highly unlikely because ML package authors are very slow to update packages) then we could consider increasing the minimum version of `grpcio`.

---

Unfortunately, this version is imposed by the protos used in the [durable task library](https://github.com/microsoft/durabletask-python/blob/main/durabletask/internal/orchestrator_service_pb2_grpc.py#L11).

@cgillum  and @famarting, can we regenerate the protos with an older grpcio version? I suggest we pin it to a version we agree on in the requirements file. I'll check for any CVEs and share here, so we can decide which version would make sense.

---

Yes let's please do that -- you should not use the latest grpcio-tools to generate the protos -- instead use an older version for wider compatibility with more versions of grpcio.

Is this only an issue in the "fork" or also the upstream version?

---

It's coming from the upstream, here's the commit: https://github.com/microsoft/durabletask-python/pull/31.

---

if we have to update the protos, I would suggest doing it here https://github.com/dapr/durabletask-protobuf

---

Here's the PR: https://github.com/dapr/durabletask-python/pull/6
Once that's merged I'll publish a new version and we can update this PR.